### ISSUE TITLE
[DataGridPro] Toolbar menus appears under the grid when closing

### ISSUE THREAD

**Issue by lucacampana-whs** [mui-org/material-ui-x#3065]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Rendering the grid, when the "density" and "export" buttons are clicked a menu shows up, when that menu is closed (clicking anywhere) it will appear for a fraction of second under the grid/at the bottom of the page

https://user-images.githubusercontent.com/82578866/139686270-caba8925-dea9-415e-b33b-e06f7519202a.mov

### Expected behavior 🤔

The menus should not move while disappearing

### Steps to reproduce 🕹

Steps:

1. Clone https://github.com/lucacampana-whs/mui-grid-toolbar-issue
2. Run `npm install` and `npm start` inside it

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: macOS 10.15.7
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 742.30 MB / 32.00 GB
    Shell: 5.1.4 - /usr/local/bin/bash
  Binaries:
    Node: 14.16.1 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.13 - /usr/local/bin/npm
  Managers:
    Homebrew: 3.2.15 - /usr/local/bin/brew
    Maven: 3.8.3 - /usr/local/bin/mvn
    pip3: 21.1.1 - /usr/local/bin/pip3
    RubyGems: 3.0.3 - /usr/bin/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 10.15. - /usr/bin/gcc
    Git: 2.24.3 - /usr/bin/git
    Clang: 12.0.0 - /usr/bin/clang
    FFmpeg: 4.4 - /usr/local/bin/ffmpeg
  Servers:
    Apache: 2.4.41 - /usr/sbin/apachectl
  Virtualization:
    Docker: 20.10.5 - /usr/local/bin/docker
  IDEs:
    Nano: 2.0.6 - /usr/bin/nano
    VSCode: 1.60.0 - /usr/local/bin/code
    Vim: 8.1 - /usr/bin/vim
    Xcode: /undefined - /usr/bin/xcodebuild
  Languages:
    Bash: 5.1.4 - /usr/local/bin/bash
    Go: 1.15.8 - /usr/local/bin/go
    Java: 1.8.0_201 - /usr/bin/javac
    Perl: 5.18.4 - /usr/bin/perl
    PHP: 7.3.11 - /usr/bin/php
    Protoc: 3.15.8 - /usr/local/bin/protoc
    Python: 2.7.16 - /usr/bin/python
    Python3: 3.9.5 - /usr/local/bin/python3
    Ruby: 2.6.3 - /usr/bin/ruby
  Databases:
    PostgreSQL: 13.2 - /Library/PostgreSQL/13/bin/postgres
    SQLite: 3.28.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 89.0.4389.72 (Used)
    Firefox: 86.0.1 (Used)
    Safari: 13.1.3
```
</details>

**Comment by flaviendelangle** [comment]
Duplicates of #2720

It was fixed in the v5.0.0-beta.3. 
Are you able to migrate to v5 ? It is as stable or more than the v4 stable release and we are strongly advising people to migrate to this version. :+1:

**Comment by lucacampana-whs** [comment]
Luckily we were able to migrate and that did the trick, thanks a lot @flaviendelangle!
